content/_guides/sasl/emacs-erc.md

```diff
@@ -0,0 +1,156 @@
+```
+;; ======== ERC START ==============================
+
+;; Most of these are just for convenience options. I believe the only
+;; absolutely necessary one here is the my-erc-connect() function I
+;; define for connecting.
+
+;; for auto-joining networks and channels when starting up ERC
+(require 'erc-join)
+(erc-autojoin-enable)
+
+;; Logging of conversations (mostly for when I forget to save peoples’
+;; answers when I ask a programming question).
+
+;; turn on logging
+(setq erc-log-mode t)
+;; write to log file when new text is added to a logged ERC buffer.
+(setq erc-log-write-after-insert t)
+;; write to log file after every message I send.
+(setq erc-log-write-after-send t)
+;; Include network name in the buffers
+(setq erc-rename-buffers t)
+
+;; from local erc-log.el and tweeked
+;; see also https://www.emacswiki.org/emacs/ErcLogging
+(require 'erc-log)
+(files--ensure-directory "~/.config")
+(files--ensure-directory "~/.config/erc")
+(files--ensure-directory "~/.config/erc/logs")
+(setq erc-log-channels-directory "~/.config/erc/logs/")
+(erc-log-enable)
+
+;; don't ask me for my nickserv password
+(setq erc-prompt-for-nickserv-password nil)
+
+;; A list of channels to auto-join
+(setq erc-autojoin-channels-alist
+	'(("irc.libera.chat" "#emacs" "#guile" "#scheme" "#erc" "#godot")))
+;; OR, if NONE..
+;; (setq erc-autojoin-channels-alist nil)
+
+;; from http://www.john2x.com/emacs.html
+;; Set the prompt to use the channel name.
+(setq erc-prompt  (lambda () (concat (buffer-name) "> ")))
+
+;; beautify the text on screen
+(setq erc-fill-function 'erc-fill-static)
+(setq erc-fill-static-center 10)
+
+;; my main function for connecting to libera.chat
+;; (see also https://libera.chat/guides/connect for regional hostnames)
+(defun my-erc-connect ()
+  (interactive)
+  (erc-tls :server "irc.libera.chat" :port 6697 :nick "MYNICK" :full-name "MY NAME"))
+(global-set-key (kbd "C-c e") 'my-erc-connect)
+
+;; Channel-tracking: Convenience functions for easily flipping between channels you are logged into
+;; from https://www.emacswiki.org/emacs/ErcChannelTracking#toc8
+;; (I don't use so much anymore, actually..)
+(defvar erc-channels-to-visit nil
+    "Channels that have not yet been visited by erc-next-channel-buffer")
+(defun erc-next-channel-buffer ()
+  "Switch to the next unvisited channel. See erc-channels-to-visit"
+  (interactive)
+  (when (null erc-channels-to-visit)
+    (setq erc-channels-to-visit 
+          (remove (current-buffer) (erc-channel-list nil))))
+  (let ((target (pop erc-channels-to-visit)))
+    (if target 
+        (switch-to-buffer target))))
+(global-set-key (kbd "C-c n") 'erc-next-channel-buffer)
+;; ======== ERC END ================================
+```
+3. In the my-erc-connect command you've defined, replace MYNICK with your nickname you plan to use (assuming no one else uses and has registered this nick).
+4. Either shut down your emacs and restart, or evaluate all the above lines.
+
+## B. Register
```

Comment:
I don't think this section should be included, I'd rather just link to `/guides/registration` in the first part and not include the other steps - this is so we keep the maintenance surface slim in the repo and don't forget to update here if something changes. It's also inconsistent with all other sasl guides right now

Reply:
I have restructured the page, and made this section a one-liner.

But this actually opens up a larger issue with the guides pages. IMHO, the overall structure of the docs is unclear for newbies.

The problem is that to set up things with SASL on LC, new users need to go through a 6-step process (as below), and that process is not clearly listed currently:

a) Set up your client for connection to libera.chat
b) Log in to libera.chat for the first time, unregistered and over a non-VPN connection
c) Use this connection opportunity to register a permanent nick and password (/_guides/registration.md)
d) Disconnect
e) Reconfigure user's client both i) resetting it for SASL login and ii) entering the new credentials
f) Reconnect and Test, logging in with new SASL client config and credentials

The individual /_guides/sasl/*.md docs pages should all make it clear that these are the 6 main steps, and then organize their sections on each page along these 6 -- just as I am doing.

For instance, the individual /_guides/sasl/*.md doc pages could have the responsibility of laying out steps a, b, d, e, and f. And for step c they could all just link to the generic /guides/registration page (as you rightly suggest, and as I now do).

But also, the /_guides/registration page should make it clear that it is *only one part* -- part "c" -- of a 6-step process, and refer users back to the sasl list page (/_guides/sasl.md).

Also, I would not agree that my page here for Emacs ERC is inconsistent with all other sasl guides right now. The offline Konversation client page (at http://userbase.kde.org/Konversation/Configuring_SASL_authentication) has a whole paragraph in parentheses in Step 9 (starting with "A short interlude:", and containing, "this guide assumes that you already have a Libera Chat account"..). Yes, this page is on a separate server, but the point is still the same: That the author still felt the need to fill in the assumption (that one has to do steps "a" and "b" above).

Since it's a bit of a maze for a new user to do all these steps, I really do think all the docs should really be re-worked and this 6-step process made explicit on every guide page, and with this ERC sasl guide I am providing a model for how to do it.